### PR TITLE
Psammead test helpers match snapshot async

### PR DESCRIPTION
**Pull request by pjlee11**
Arises from https://github.com/bbc/simorgh/pull/4240/

As a POC the logic in this PR was added [here](https://github.com/bbc/simorgh/blob/dd379421fb5b3f12c57d82b0633b2fc50724b77b/src/app/containers/CpsAssetPageMain/testHelpers.js#L1) to prove we could have snapshot tests generated from async it() blocks. 

**Code changes:**

- Creates `createSnapshot` as a refactor to share the snapshot generating code. 
- Adds `matchSnapshotAsync` which is to be used inside of an `async` it() block. This is required because the execution of the async it must be in Simorgh or the describe block is not able to await the test being run. 

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated jest tests added (for new features) or updated (for existing features)
- ~[ ] This PR requires manual testing~

### CODE REVIEW

File: packages/utilities/psammead-test-helpers/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 <!-- prettier-ignore -->
 | Version | Description |
 |---------|-------------|
+| 3.0.3 | [PR#xxxx](https://github.com/bbc/psammead/pull/xxxx) Adds matchSnapshotAsync export |
```
Comment (pjlee11) [inline packages/utilities/psammead-test-helpers/CHANGELOG.md]:
```suggestion
| 3.0.3 | [PR#2416](https://github.com/bbc/psammead/pull/2416) Adds matchSnapshotAsync export |
```

File: packages/utilities/psammead-test-helpers/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@bbc/psammead-test-helpers",
-  "version": "3.0.2",
+  "version": "3.0.3",
```
Comment (dr3) [inline packages/utilities/psammead-test-helpers/package.json]:
```suggestion
  "version": "3.1.0",
```

**Review by dr3 — CHANGES_REQUESTED** [review]
LGTM, just version tweak

**Review by dr3 — APPROVED** [review]

**Review by rhenshaw56 — APPROVED** [review]
Looks good! most of the lines aren't covered though

File: packages/utilities/psammead-test-helpers/index.test.jsx [hunk 3]
```diff
@@ -232,4 +233,14 @@ describe('Psammead test helpers', () => {
     'should match the snapshot for the test component with helmet and other content',
     <HelmetWithContent />,
   );
+
+  it('should create a snapshot from an async it() block with component and done props', async done => {
+    const data = Promise.resolve('Foobar');
```
Comment (jroebu14) [inline packages/utilities/psammead-test-helpers/index.test.jsx:238]: I think this should be `const data = await Promise.resolve('Foobar');`

File: packages/utilities/psammead-test-helpers/index.test.jsx [hunk 4]
```diff
@@ -232,4 +233,14 @@ describe('Psammead test helpers', () => {
     'should match the snapshot for the test component with helmet and other content',
     <HelmetWithContent />,
   );
+
+  it('should create a snapshot from an async it() block with component and done props', async done => {
```
Comment (jroebu14) [inline packages/utilities/psammead-test-helpers/index.test.jsx:237]: you don't need to call done when using async/await https://jestjs.io/docs/en/tutorial-async#async-await

Comment (jroebu14) [reply packages/utilities/psammead-test-helpers/index.test.jsx:237]: I didn't think it was possible to use async/await in Psammead https://github.com/bbc/psammead/issues/2126

I haven't tried this release yet but I suspect this package is broken if the above issue hasn't been sorted

Comment (pjlee11) [reply packages/utilities/psammead-test-helpers/index.test.jsx:237]: As the async/await is only in the tests it shouldn't be affected by #2126

**Pull request merged**